packages/firestore/src/api/database.ts

```diff
@@ -1354,6 +1355,15 @@ export class Query implements firestore.Query {
     validateExactNumberOfArgs('Query.where', arguments, 3);
     validateArgType('Query.where', 'non-empty string', 2, opStr);
```

Comment:
As mentioned below, I think we could drop this and rely on the new validation function to handle this case.